synapse/app/client_reader.py

```diff
@@ -84,6 +85,7 @@ class ClientReaderSlavedStore(
     SlavedTransactionStore,
     SlavedProfileStore,
     SlavedClientIpStore,
+    SlavedUserDirectoryStore,
```

Comment:
TBH I'd be tempted to just inherit from `UserDirectoryStore` directly, since `SlavedUserDirectoryStore` doesn't really do anything. YMMV

Reply:
I followed the pattern of the `DirectoryStore` which does the same thing: https://github.com/matrix-org/synapse-dinsic/blob/85845e048cd8d2674ad38dd00e8a934b4b960adc/synapse/replication/slave/storage/directory.py#L16-L22

Happy to simplify it though if you'd like!